src/pages/map/map.spec.ts

```diff
@@ -157,107 +168,145 @@ describe('MapPage', function () {
       fixture.detectChanges();
       await fixture.whenStable();
 
-      expect(spyFetchAll.callCount).to.equal(2);
-      expect(spyFetchAll.getCall(1).args[0]).to.eql(callOptions);
+      expect(spyFetchAll).to.have.been.calledTwice;
+      expect(spyFetchAll).to.have.been.calledWith(callOptions);
 
       markerIds = component.layers.map(marker => marker.id);
       expectedIds = locationsDataMock.slice(1).map(location => location.id);
 
       expect(markerIds).to.have.members(expectedIds);
     });
 
-    describe('and initialized', function() {
+    it('should open a Popover when a location\'s marker is clicked', async function () {
+      const spyFetchOne = spy(component.locationsService, 'fetchOne');
+
+      const eMock = {
+        target: {
+          id: 'c821bc0f-85b4-44d5-9bbe-a30cf197c30a',
+          _latlng: new L.LatLng(56, 9)
+        }
+      }
+
+      const responses = [
+        { body: JSON.stringify(locationsDataMock) },
+        { body: JSON.stringify(locationsDataMock[0]) }
+      ];
+
+      backend.connections.subscribe((connection: any) => {
+        if (connection.request.url.includes(`${MockEnv.backendUrl}/locations`)) {
+          connection.mockRespond(new Response(new ResponseOptions(responses.shift())));
+        } else if (connection.request.url.includes(`${MockEnv.backendUrl}/locations/c821bc0f-85b4-44d5-9bbe-a30cf197c30a`)) {
+          connection.mockRespond(new Response(new ResponseOptions(responses.shift())));
+        }
+      });
+
+      fixture.detectChanges();
+      await fixture.whenStable();
+
+      expect(spyFetchOne).to.have.not.been.called;
+      expect(responses).to.have.lengthOf(1);
+
+      component.onLocationClicked(eMock);
+
+      expect(popoverCtrlMock.create).to.have.been.calledOnce;
+      expect(popoverCtrlMock.create.args[0][0]).to.eql(LocationDetails);
+      expect(popoverCtrlMock.create.args[0][1]).to.eql({ locationId: eMock.target.id });
```

Comment:
Also check the [third argument](https://github.com/MediaComem/biopocket-mobile/blob/story-poi-details/src/pages/map/map.ts#L137-L141).

When checking a spy's arguments, prefer using `eql` to check the entire array when possible (e.g. https://github.com/MediaComem/biopocket-mobile/blob/sprint-1-discovering-biodiversity/src/pages/map/map.spec.ts#L202), to avoid forgetting arguments.

If the entire array cannot be compared with `eql` (e.g. some objects cannot be automagically compared), add an additional check on the length of the array.